agentcfg/model.go

```diff
@@ -18,88 +18,81 @@
 package agentcfg
 
 import (
-	"crypto/md5"
 	"encoding/json"
 	"fmt"
-	"hash"
-	"sort"
-	"strings"
 )
 
-// Doc represents an elasticsearch document
-type Doc struct {
-	Settings Settings
-	ID       string
+const (
+	// ServiceName keyword
+	ServiceName = "service.name"
+	// ServiceEnv keyword
+	ServiceEnv = "service.environment"
+	// Etag / If-None-Match keyword
+	Etag = "ifnonematch"
+)
+
+var (
+	// RumAgent keywords (new and old)
+	RumAgent = []string{"rum-js", "base-js"}
+	// RumSettings are whitelisted applicable settings for RUM
+	RumSettings = []string{"transaction_sample_rate"}
+)
+
+// Result models a Kibana response
+type Result struct {
+	Source Source `json:"_source"`
 }
 
-// Settings hold agent configuration
-type Settings map[string]string
+// Source is the Elasticsearch _source
+type Source struct {
+	Settings Settings `json:"settings"`
+	Etag     string   `json:"etag"`
+	Agent    string   `json:"agent_name"`
+}
 
-// NewDoc unmarshals given byte slice into a Doc instance
-func NewDoc(inp []byte) (*Doc, error) {
-	settings, err := unmarshal(inp)
-	if err != nil {
-		return nil, err
-	}
+// Query represents an URL body or query params for agent configuration
+type Query struct {
+	Service Service `json:"service"`
+	Etag    string  `json:"etag"`
+	IsRum   bool    `json:"-"`
+}
 
-	h := md5.New()
-	var out = map[string]string{}
-	if err := parse(settings, out, "", h); err != nil {
-		return nil, err
-	}
+func (q Query) toString() string {
```

Comment:
I'd prefer calling this `ID` as that's better describing what it does, and `toString` is usually used for a nicer presentation of an instance.